tree/treeplayer/src/TTreeReaderArray.cxx

```diff
@@ -475,12 +475,13 @@ void ROOT::Internal::TTreeReaderArrayBase::CreateProxy()
       if (branch->GetTree() != fTreeReader->GetTree()->GetTree()) {
          // It is in a friend, let's find the 'index' in the list of friend ...
          int index = -1;
-         int current = 0;
-         for(auto fe : TRangeDynCast<TFriendElement>( fTreeReader->GetTree()->GetTree()->GetListOfFriends())) {
-            if (branch->GetTree() == fe->GetTree()) {
-               index = current;
-            }
-            ++current;
+         const auto friendElements =
```

Comment:
Why is the new code 'better'? (it seems harder to read/more characters).

Side note (likely in practice irrelevant): the 2 code seems to not do exactly the same things if the same tree is twice in the list of friends.

Reply:
The change uses more modern C++ syntax (and subjectively is more readable to me). I also believe it introduces a fix regarding what you are mentioning: it stops the traversal of the friends list earlier thus is less expensive and avoids picking an eventual second index as the index for this tree.

As a side note, if the situation where the exact same `TTree *` could potentially be used as friend multiple times for the same main tree, then neither the current version nor the previous version of the code would be correct, and that would be a bug that should be addressed separately (outside of this PR)

Reply:
The equivalent code in `TTreeReaderValue.cxx:609` has a `break` that make the 2 code (old and std algorithm) strictly equivalent (slight potential difference in number of arithmetic operations).

It is a bit of a matter of taste but I (personally) still find the old code 'clearer' as the "matching criteria" is clearly on its own line:
```
    if (branch->GetTree() == fe->GetTree()) {
```
while in the new code, it is buried in the middle of a 4 lines statements (and at the end of line)
```
         if (auto foundFriend =
                std::find_if(friendElements.begin(), friendElements.end(),
                             [&branch](TFriendElement *fe) { return branch->GetTree() == fe->GetTree(); });
```